force-app/main/default/classes/RD2_ERecurringDonationsSelector.cls

```diff
@@ -50,27 +54,35 @@ public inherited sharing class RD2_ERecurringDonationsSelector {
      * @description Returns the Recurring Donation records related to the Experience Site user.
      * @return List<npe03__Recurring_Donation__c>
      */
-    public List<npe03__Recurring_Donation__c> getRecurringDonations() {
+    public List<npe03__Recurring_Donation__c> getRecurringDonations(String elevateFilter) {
```

Comment:
Have we considered using dynamic SOQL here? That way we are not having to loop through all the RDs returned and filter based on `CommitmentId__c`.

Reply:
Hi @voduyemi , thank you very much for your feedback. I updated this method based on your feedback, Can you check it again?